### ISSUE TITLE
alias domains are not shown to domain admins

### ISSUE THREAD

**Issue by benjyamon** [postfixadmin/postfixadmin#149]
unless we give them admin privileges over the alias domain(s) as well. But doing that has drawbacks like opening the door to user confusion as they might attempt to add mailboxes (or aliases) to the **alias** domain (which we can ultimately prevent by setting the maximums to -1) because it now shows up as a domain they can admin just like the real / primary domain.

The old version 2.3.7 didn't behave this way - it did show domain admins their alias domains.

Do you think it might be possible to go back to showing the alias domains to domain admins?

I've been trying to figure out the code (xdebug: list-virtual.php --> PFAHandler.php --> functions.php list_domains_for_admin())  so I could do more than just file an issue here (but offer the fixes too!), but I'm getting a bit too confused.

**Comment by DavidGoodwin** [comment]
(You're not alone finding PFAHandler etc hard to follow ... I think only @cboltz really understands it).

**Comment by cboltz** [comment]
Actually this behaviour change is intentional ;-)

In 2.3, admins that had access to "one half" of an alias domain could change or delete it, but it was impossible to restore the original target unless they had permissions for the target domain.
Since this is highly annoying (you need to call a superadmin after accidently clicking on "delete"), I decided that it's a better idea not to allow such accidents ;-)

If you really want to change that, have a look at initStruct() in model/AliasdomainHandler. You'll probably need to add more domains to `struct['target_domain']['options']` (untested). (list-virtual.php shouldn't need a change.) 

I'm not sure if I should recommend this solution for the reasons given above. Allowing access to the target domain and setting the limits to -1 is the better way IMHO.

BTW: Once David fixes the whitespace breakage the automated code formatting introduced, the initStruct() function will be readable again ;-)

**Comment by DavidGoodwin** [comment]
(I've not forgotten ... )

**Comment by benjyamon** [comment]
thanks for responding! i think (where we need to) we will just allow postmasters access to their alias domains (and -1 limits). 
closing!
